python/imageutils.py

```diff
@@ -0,0 +1,481 @@
+                    avg_seek = 0
+                    avg_write = 0
+                    min_seek = 1000
+                    min_write = 1000
+                    std_seek = 0
+                    std_write = 0
+    
+                    seek_times = []
+                    write_times = []
+
+                    step_x = 1
+                    step_z = 1
+
+                    start_x = 0
+                    start_z = 0
+
+                    end_x = shape_x
+                    end_z = shape_z
+                    
+                    #print "Write starting at position: ({0}, {1}, {2})".format(block_y, block_z + start_z , block_x + start_x)
+
+                    for i in range(start_x, end_x, step_x):
+                        for j in range(start_z, end_z, step_z):
+                            seek_start = time()
+                            reconstructed.seek(self.header_size + bytes_per_voxel*(block_y + (block_z + j)*y_size +(block_x + i)*y_size*z_size), 0)
+                            seek_end = time() - seek_start
+                            write_start = time()
+                            reconstructed.write(block_data[:,j, i].tobytes())
+                            write_end = time() - write_start
+
+                            avg_seek += seek_end
+                            avg_write += write_end 
+
+                            seek_times.append(seek_end)
+                            write_times.append(write_end)
+
+                            if seek_end > max_seek:
+                                max_seek = seek_end
+                            if write_end > max_write:
+                                max_write = write_end
+                            if seek_end < min_seek:
+                                min_seek = seek_end
+                            if write_end < max_write:
+                                min_write = write_end
+
+
+                    #print time()-t
+                    
+                    #avg_seek = avg_seek / (shape_x * shape_z)
+                    #avg_write = avg_write / (shape_x * shape_z)
+                    
+                    #print "Maximum seek time: {0}\t Maximum write time: {1}".format(max_seek, max_write)
+                    #print "Minimum seek time: {0}\t Minimum write time: {1}".format(min_seek, min_write)
+                    #print "Average seek time: {0}\t Average write time: {1}".format(avg_seek, avg_write)
+                    #print "Std seek: {0}\t Std write: {1}".format(np.std(seek_times), np.std(write_times))
+
+                    #if end_x == -1:
+                    #    end_x = 0
+                    #if end_z == -1:
+                    #    end_z = 0
+                    #prev_b_end = (block_y + shape[0], block_z + end_z, block_x + end_x)
+                    #print "End position of write: {0}".format(prev_b_end)
+    
+    #will only work if the amount of blocks that fit in memory represents a rectangular prism
+    def cp_block_slice(self, legend, mem):
```

Comment:
A line of comments on each parameter would be useful (in particular, what is the unit of mem?)